server/src/test/java/org/elasticsearch/index/mapper/TimeSeriesIdFieldMapperTests.java

```diff
@@ -720,38 +720,4 @@ public void testParseWithDynamicMappingInvalidRoutingHash() {
         });
         assertThat(failure.getMessage(), equalTo("[5:1] failed to parse: Illegal base64 character 20"));
     }
-
-    public void testParseWithDynamicMappingNullId() {
```

Comment:
This would hit the assert, it's not possible to reach the case where both the id and routing are null and the exception is thrown.

Good point though, restored and modified the test to cover the new logic.